frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install.tsx

```diff
@@ -32,31 +41,52 @@ export const CreateAttachedDevicesCluster: React.FC<CreateAttachedDevicesCluster
   const [isLsoPresent, setIsLsoPresent] = React.useState(false);
   const [allDataLoaded, setAllDataLoaded] = React.useState(false);
   const [lsoNs, setLsoNs] = React.useState('');
+  const [lsoVersion, setLsoVersion] = React.useState('');
+
+  const subscriptionResource: WatchK8sResource = {
+    kind: referenceForModel(SubscriptionModel),
+    isList: true,
+  };
   const [subscription, subscriptionLoaded, subscriptionLoadError] = useK8sWatchResource<
     K8sResourceKind[]
-  >(LSOSubscriptionResource);
+  >(subscriptionResource);
+
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    fieldSelector: `metadata.name=${lsoVersion}`,
+    namespace: lsoNs,
+    isList: true,
+  };
+  const [csv, csvLoaded, csvLoadError] = useK8sWatchResource<K8sResourceKind[]>(csvResource);
 
   React.useEffect(() => {
-    if (subscriptionLoadError || (!subscription.length && subscriptionLoaded)) {
+    const lsoSubscription = subscription.filter((item) => item?.spec?.name === LSO_OPERATOR);
+    setLsoNs(getNamespace(lsoSubscription[0]));
+    setLsoVersion(getOperatorVersion(lsoSubscription[0]));
```

Comment:
-> I am doing filtering for a LSO subscription from the list of all installed subscriptions and I am doing this inside `useEffect` hook (line 63) so that we do not have to loop over all the subscriptions again in case component re-renders, we will only do so if subscription/csv resource changes.
-> Once we get the first LSO operator (line 63) , now when the `CreateAttachedDevicesCluster` FC re-renders this will be passed to CSV resource (line 56 & 57) and we will get the CSV corresponding to this specific filtered LSO operator.
-> Once we get the LSO subscription and the corresponding CSV, `useEffect` will not be called any more and also value of `lsoNs` and `lsoVersion` will not change.
-> If we extract ns and version outside the hook then I have to declare `lsoSubscription` variable of type either `let` or `var` outside of `CreateAttachedDevicesCluster` FC otherwise value of the variable ` lsoSubscription` will reset to default (initialised during variable declaration) every time FC will re-render because of the watch-hook. 
Or, I have to use `useState` hook for `lsoSubscription` instead of `lsoNs` and `lsoVersion`.

This is why I used set state. I might have missed something, so, I am looking forward to your feedback on my comment.

Reply:
I guess there was some misunderstanding from my end... I did not see `var` and `let` variables used at that many places in our code so I thought may be we only use `const` in practice as much as possible...
Just to confirm, is it okay to use `var` or `let`?

Reply:
okay... will fix that...